### PR TITLE
Fix go vet warnings

### PR DESCRIPTION
**Pull request by jen20**
Continues and rebases #5122

### CODE REVIEW

File: terraform/state_test.go [hunk 1]
```diff
@@ -795,9 +795,6 @@ func TestReadWriteState(t *testing.T) {
 		},
 	}
 
-	// Checksum before the write
```
Comment (jen20) [inline terraform/state_test.go]: @phinze this is the change we discussed.

**Comment by phinze** [comment]
Wow these are some nice catches. LGTM

**Comment by tpounds** [comment]
@jen20 Thanks for picking this up! What do you guys think about enabling all warnings and always the vet target?

**Comment by jen20** [comment]
@tpounds - I think we should definitely vet in the target that builds on Travis and fail the build if there are any warnings. IIRC removing the arguments to vet should achieve that?

**Pull request merged**